library/src/scala/quoted/matching/Lambda.scala

```diff
@@ -0,0 +1,31 @@
+package scala.quoted
+package matching
+
+/** Lambda expression extractor */
+object Lambda {
+
+  /** `case Lambda(body)` matche a lambda and extract the body.
```

Comment:
```suggestion
  /** `case Lambda(fn)` matches a lambda by lifting the function from `S => T` to `Expr[S] => Expr[T]`.
```

It seems better to use  `fn` as the name to avoid misunderstanding.